### DISCUSSION TITLE
Detekt in Android project: How can I enable it for xml layout files?

### DISCUSSION THREAD

**Discussion by Pabloku** [detekt/detekt#4956] [Q&A] [answered]
Hi!

I'd like to create a custom rule to check that every activity layout xml file has a property defined in its parent container. To be able to do that, first I need Detekt to be run over src/main/res/layout/*.xml files. I tried using setSource in my detekt.gradle file but it didn't work.
Any ideas on how to do this?

Thanks very much,

Pabloku

**Comment by cortinico** [comment, marked as answer]
You can't do this with Detekt. You'll have to use Android Lint as you need to access an Android Resource

**Reply by Pabloku** [reply]
Thanks for clarification!

**Comment by IvanSanchez-uam** [comment]
is it the same for gradle file like properties or settings ?

**Reply by schalkms** [reply]
Please take a look at my reply here:
https://github.com/detekt/detekt/discussions/5477#discussioncomment-3964442
